Orange/canvas/document/editlinksdialog.py

```diff
@@ -782,3 +864,74 @@ def setDocument(self, doc):
     def _onDocumentSizeChanged(self, size):
         """The doc size has changed"""
         self.updateGeometry()
+
+
+class LinkLineItem(QGraphicsLineItem):
+    """
+    A line connecting two Channel Anchors.
+    """
+
+    def __init__(self, parent=None):
+        QGraphicsLineItem.__init__(self, parent)
+        self.setAcceptHoverEvents(True)
+
+        self.__shape = None
+
+        self.__default_pen = QPen(QColor('#383838'), 4)
+        self.__default_pen.setCapStyle(Qt.RoundCap)
+        self.__hover_pen = QPen(QColor('#000000'), 4)
+        self.__hover_pen.setCapStyle(Qt.RoundCap)
+        self.setPen(self.__default_pen)
+
+        self.__shadow = QGraphicsDropShadowEffect(
+            blurRadius=10, color=QColor('#9CACB4'),
+            offset=QPointF(0, 0)
+        )
+
+        self.setGraphicsEffect(self.__shadow)
+        self.prepareGeometryChange()
+        self.__shadow.setEnabled(False)
+
+    @overload
+    def setLine(self, line: QLineF) -> None:
+        pass
+
+    @overload
+    def setLine(self, x1: float, y1: float, x2: float, y2: float) -> None:
```

Comment:
Such use of `overload` is only valid since python 3.5.2 (note the micro release). It raises on error on Python<=3.5.1. I would avoid its use or put it in a `if False:` block.

Reply:
Never mind. It seems we are on longer even syntax compatible with Python 3.5 (since gh-3403).

Reply:
Do these overloads have any effect? PyCharm and Pylint don't recognize them, and the redefinitions trigger `function-redefined` in pylint.

If they are necessary, please disable pylint warning. If they have no effect, please remove them.

Reply:
They're not necessary, but PyQT already overloads the method. I thought it would be most elegant to expose the same interface if I am overriding the method.